src/pip/_internal/cli/cmdoptions.py

```diff
@@ -25,6 +26,25 @@
     from typing import Any  # noqa: F401
 
 
+class OptionValues(optparse.Values):
```

Comment:
Would a cleaner option be to have a custom action callback function instead? It's definitely lighter weight than a subclass which has to be used everywhere.

https://docs.python.org/3/library/optparse.html#optparse-option-callbacks